ffi/gn/BUILDCONFIG.gn

```diff
@@ -0,0 +1,39 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+if (target_os == "") {
+  target_os = host_os
+}
+if (target_cpu == "") {
+  target_cpu = host_cpu
+}
+if (current_cpu == "") {
+  current_cpu = target_cpu
+}
+if (current_os == "") {
+  current_os = target_os
+}
+
+_configs = [ "//build:rust_defaults" ]
+
+set_defaults("executable") {
+  configs = _configs
+}
+set_defaults("rust_library") {
+  configs = _configs
+}
+
+# ==============================================================================
+# TOOLCHAIN SETUP
+# ==============================================================================
+#
+# Here we set the default toolchain, as well as the variable host_toolchain
+# which will identify the toolchain corresponding to the local system when
+# doing cross-compiles. When not cross-compiling, this will be the same as the
+# default toolchain.
+
+host_toolchain = "//build/rust_host"
+
+current_cpu = "wasm32"
+set_default_toolchain("//build/rust_wasm")
```

Comment:
@zanderso What would be the best practices for setting the `current_cpu` and `current_os` variables and making sure they get passed down to the `--target` flag in rustc invocations?

Reply:
You can create a GN template to generate toolchain definitions that is parameterized by the target OS and architecture. Then in BUILDCONFIG.gn, you can pass the right generated definition to `set_default_toolchain`.

Here's an example template that Flutter uses to generate C/C++ toolchains: https://github.com/flutter/buildroot/blob/master/build/toolchain/gcc_toolchain.gni.

Here is how it's applied for Linux: https://github.com/flutter/buildroot/blob/master/build/toolchain/linux/BUILD.gn

And here is how the BUILDCONFIG.gn selects based on `current_cpu` etc: https://github.com/flutter/buildroot/blob/master/build/config/BUILDCONFIG.gn#L543

Reply:
Yep, thanks! I'm doing this now. I'm trying to figure out the minimal setup.